app/assets/javascripts/visualizations/highvis/box.coffee

```diff
@@ -0,0 +1,350 @@
+              upperBound = q3
+
+          else if @configs.whiskerMode == "max-min"
+            lowerBound = min
+            upperBound = max
+
+          lowerOutliers = true
+          upperOutliers = true
+          if min >= lowerBound
+            lowerBound = min
+            lowerOutliers = false
+          if max <= upperBound
+            upperBound = max
+            upperOutliers = false
+
+          thisBox = {
+            x: index
+            low: lowerBound
+            q1: q1
+            median: median
+            q3: q3
+            high: upperBound
+            color: globals.getColor(groupIndex)
+            name: data.groups[groupIndex] or data.noField()
+            lowerOutliers: lowerOutliers
+            upperOutliers: upperOutliers
+          }
+
+          boxes.push(thisBox)
+          outliers = []
+
+          if lowerOutliers or upperOutliers
+            for point in data.selector(@configs.displayField, groupIndex, dp)
+              if point < lowerBound or point > upperBound
+                outliers.push([index, point])
+            allOutliers.push({gindex: groupIndex, points: outliers})
+              
+          index++
+
+        gcolors = []
+        groups = []
+        for g, gi in data.groups when gi in data.groupSelection
+          gcolors.push(globals.getColor(gi))
+          groups.push(g)
+
+        if(boxes.length > 0)
+          pointWidth = null
+          if boxes.length < 3
+            pointWidth = 250 # limit the size of boxes when only a few a present
+
+          boxSeries = {
+            data: boxes
+            type: "boxplot"
+            showInLegend: false
+            pointWidth: pointWidth
+            lineWidth: 3
+          }
+          @chart.addSeries boxSeries, false
+
+        for o in allOutliers
+          outlierSeries = {
+            name: data.groups[o.gindex] or data.noField()
+            type: 'scatter'
+            data: o.points
+            color: globals.getColor(o.gindex)
+            marker:
+              symbol: "circle"
+            showInLegend: false
+            field: fieldTitle(data.fields[@configs.displayField])
+            fieldUnit: fieldUnit(data.fields[@configs.displayField], false)
+          }
+          @chart.addSeries outlierSeries, false
+
+        if @configs.meanLine
+          # Update the mean line
+          @chart.yAxis[0].removePlotLine("meanLine")
+          mean = data.getMean(@configs.displayField, data.groupSelection, dp)
+          @chart.yAxis[0].addPlotLine({
+            color: 'red'
+            width: 2
+            value: mean
+            zIndex: 10 # draw on top
+            id: 'meanLine'
+            label:
+              text: 'Mean: ' + mean
+              style:
+                color: 'gray'
+          })
+
+        @chart.redraw()
+
+
+
+      buildLegendSeries: ->
+        for groupIndex in data.groupSelection.sort()
+          options =
+            # dummy series needs to be scatter, otherwise it pushes
```

Comment:
Matt is having a problem similar to this. Can you look at it and see if you have any ideas.